rpi: Do not build firmware if MACHINE is not raspberry pi.

masami256 opened the pull request:
# Purpose of pull request

linux-firmware-rpidistro_git recipe is raspberry pi specific recipe.
So, other machines(e.g. qemuarm64) don't need to build it.
This commit fixes build error when running bitabke world with MACHINE=qemarm64.

# Test
## How to test

for rpi3, build core-image-minimal should success
for qemuarm64, build linux-firmware-rpidistro  should fail because qemuarm64 is not in COMPATIBLE_MACHINE.

## Test result

### raspberrypi3-64

```
masami@ubuntu1804:~/emlinux/build$ bitbake core-image-minimal
Parsing recipes: 100% |########################################################################################################################################################################################################| Time: 0:00:38
Parsing of 1338 .bb files complete (0 cached, 1338 parsed). 2356 targets, 69 skipped, 4 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-18.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "raspberrypi3-64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.5"
TUNE_FEATURES        = "aarch64 cortexa53 crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "warrior:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "warrior:aecbfc0d5c84cb48c4f08dfa8534442631b7fd8a"
meta-debian-extended = "warrior:bbb54b3d23659529b106993a87f73493c49468ef"
meta-emlinux         = "update-rpi-recipe:00070a7282da00065a669eabc6ef6caf582ec02e"

Initialising tasks: 100% |#####################################################################################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 853 Found 0 Missed 853 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2942 tasks of which 5 didn't need to be rerun and all succeeded.
```
And, linux-firmware-rpidistro recipe copied license file(LICENCE.broadcom_bcm43xx) correctly.

```
masami@ubuntu1804:~/emlinux/build$ wic ls tmp-glibc/deploy/images/raspberrypi3-64/core-image-minimal-raspberrypi3-64.wic:2/lib/firmware
debugfs 1.44.1 (24-Mar-2018)
    766   40755 (2)      0      0    1024 26-Apr-2022 09:03 .
    156   40755 (2)      0      0    1024 26-Apr-2022 09:03 ..
    767   40755 (2)      0      0    1024 26-Apr-2022 09:03 brcm
    771  100644 (1)      0      0    4178 26-Apr-2022 09:02 LICENCE.broadcom_bcm43xx
```
### qemuarm64

Build error occurs as expected.

```
masami@ubuntu1804:~/emlinux/build$ MACHINE=qemuarm64 bitbake linux-firmware-rpidistro 
Parsing recipes: 100% |########################################################################################################################################################################################################| Time: 0:00:26
Parsing of 1338 .bb files complete (0 cached, 1338 parsed). 2356 targets, 78 skipped, 4 masked, 0 errors.
ERROR: Nothing PROVIDES 'linux-firmware-rpidistro'
linux-firmware-rpidistro was skipped: incompatible with machine qemuarm64 (not in COMPATIBLE_MACHINE)

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
```
bitbake world was succeeded in qemuarm64.

```
masami@emlinuxdev:~/emlinux/truly-latest/build$ bitbake world
Loading cache: 100% |##################################################################################################################################################################| Time: 0:00:00
Loaded 2358 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-18.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.5"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "HEAD:aecbfc0d5c84cb48c4f08dfa8534442631b7fd8a"
meta-debian-extended = "HEAD:27821f6b789ab21e4b65a92fe477d8bbef4088a3"
meta-emlinux         = "update-rpi-recipe:00070a7282da00065a669eabc6ef6caf582ec02e"

Initialising tasks: 100% |#############################################################################################################################################################| Time: 0:00:11
Sstate summary: Wanted 3969 Found 0 Missed 3969 Current 1052 (0% match, 20% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
WARNING: lttng-modules-2.10.8-r0 do_package: lttng-modules: no modules were created; this may be due to CONFIG_TRACEPOINTS not being enabled in your kernel.
WARNING: libpwquality-1.4.0-r0 do_populate_lic: libpwquality: No generic license file exists for: libpwquality in any provider
WARNING: syslog-ng-3.19.1-r0 do_populate_lic: syslog-ng: No generic license file exists for: LGPL-2.1-with-OpenSSL-exception in any provider
NOTE: Tasks Summary: Attempted 14238 tasks of which 6019 didn't need to be rerun and all succeeded.

Summary: There were 3 WARNING messages shown.
```

hiraku-wfs commented on recipes-kernel/linux-firmware-rpidistro/linux-firmware-rpidistro_git.bb:
```diff
@@ -8,6 +8,7 @@ RPi-Distro obtains these directly from Cypress; they are not submitted \
 to linux-firmware for general use."
 HOMEPAGE = "https://github.com/RPi-Distro/firmware-nonfree"
 SECTION = "kernel"
+COMPATIBLE_MACHINE = "^ras"
```
Why is the pattern abbreviated? I think it is too short and will affect unexpected MACHINEs.

masami256 replied:
@hiraku-wfs  Other recipes e.g.([rpi-bootfiles.bb](https://github.com/miraclelinux/meta-emlinux/blob/warrior/recipes-bsp/bootfiles/rpi-bootfiles.bb#L16), [rpi-u-boot-scr.bb](https://github.com/miraclelinux/meta-emlinux/blob/warrior/recipes-bsp/rpi-u-boot-scr/rpi-u-boot-scr.bb)) uses this pattern.

Is it okay to change other recipes to "raspberrypi 3-64" too?

hiraku-wfs replied:
We should use SOC_FAMILY here like meta-raspberrypi.

https://github.com/agherzan/meta-raspberrypi/blob/0135a02ea577bd39dd552236ead2c5894d89da1d/conf/machine/include/rpi-base.inc
```
SOC_FAMILY = "rpi"
include conf/machine/include/soc-family.inc
```

Please add the above description into the meta-emlinux's raspberrypi3-64.conf, then we can use "^rpi$" pattern for the COMPATIBLE_MACHINE.

masami256 replied:
@hiraku-wfs Thank you the comment. I've updated recipes.

Can build core-image-minimal for rpi.

```
masami@ubuntu1804:~/emlinux/build$ bitbake core-image-minimal
Parsing recipes: 100% |########################################################################################################################################################################################################| Time: 0:00:39
Parsing of 1338 .bb files complete (0 cached, 1338 parsed). 2356 targets, 69 skipped, 4 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "ubuntu-18.04"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "raspberrypi3-64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.5"
TUNE_FEATURES        = "aarch64 cortexa53 crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "warrior:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "warrior:aecbfc0d5c84cb48c4f08dfa8534442631b7fd8a"
meta-debian-extended = "warrior:bbb54b3d23659529b106993a87f73493c49468ef"
meta-emlinux         = "update-rpi-recipe:36b4bca01f42feecb53c789879eaa7dcba1bdf7f"

Initialising tasks: 100% |#####################################################################################################################################################################################################| Time: 0:00:01
Sstate summary: Wanted 853 Found 0 Missed 853 Current 0 (0% match, 0% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 2942 tasks of which 5 didn't need to be rerun and all succeeded.
```

Cannot build raspberry pi specific recipes.

```
masami@ubuntu1804:~/emlinux/build$ MACHINE=qemuarm64 bitbake rpi-bootfiles -c fetch
Loading cache: 100% |##########################################################################################################################################################################################################| Time: 0:00:00
Loaded 841 entries from dependency cache.
Parsing recipes: 100% |########################################################################################################################################################################################################| Time: 0:00:17
Parsing of 1338 .bb files complete (452 cached, 886 parsed). 2356 targets, 78 skipped, 4 masked, 0 errors.
ERROR: Nothing PROVIDES 'rpi-bootfiles'
rpi-bootfiles was skipped: incompatible with machine qemuarm64 (not in COMPATIBLE_MACHINE)

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
masami@ubuntu1804:~/emlinux/build$ MACHINE=qemuarm64 bitbake rpi-u-boot-scr -c fetch
Loading cache: 100% |##########################################################################################################################################################################################################| Time: 0:00:00
Loaded 2356 entries from dependency cache.
ERROR: Nothing PROVIDES 'rpi-u-boot-scr'
rpi-u-boot-scr was skipped: incompatible with machine qemuarm64 (not in COMPATIBLE_MACHINE)

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
masami@ubuntu1804:~/emlinux/build$ MACHINE=qemuarm64 bitbake linux-firmware-rpidistro -c fetch
Loading cache: 100% |##########################################################################################################################################################################################################| Time: 0:00:00
Loaded 2356 entries from dependency cache.
ERROR: Nothing PROVIDES 'linux-firmware-rpidistro'
linux-firmware-rpidistro was skipped: incompatible with machine qemuarm64 (not in COMPATIBLE_MACHINE)

Summary: There was 1 ERROR message shown, returning a non-zero exit code.
```

masami256 committed:
raspberrypi: Add rpi to SOC_FAMILY

Add "rpi" as SOC_FAMILY to use soc name to set COMPATIBLE_MACHINE
variable.

Signed-off-by: Masami Ichikawa <masami.ichikawa@miraclelinux.com>

masami256 committed:
raspberrypi: Add COMPATIBLE_MACHINE condition

linux-firmware-rpidistro is raspberrypi specific recipe, so add
COMPATIBLE_MACHINE to not build on other MACHINE.

Signed-off-by: Masami Ichikawa <masami.ichikawa@miraclelinux.com>

hiraku-wfs commented:
Thanks!
